cpp/test/sg/fil_test.cu

```diff
@@ -247,9 +248,21 @@ class BaseFilTest : public testing::TestWithParam<FilTestParams> {
     CUDA_CHECK(cudaFree(mask_d));
   }
 
+  void apply_softmax(float* class_scores) {
+    float max = *std::max_element(class_scores, &class_scores[ps.num_classes]);
+    for (int i = 0; i < ps.num_classes; ++i)
+      class_scores[i] = expf(class_scores[i] - max);
+    float sum =
+      std::accumulate(class_scores, &class_scores[ps.num_classes], 0.0f);
+    for (int i = 0; i < ps.num_classes; ++i) class_scores[i] /= sum;
+  }
+
   void transform(float f, float& proba, float& output) {
     if ((ps.output & fil::output_t::AVG) != 0) {
-      f *= (1.0f / ps.num_trees);
+      if (ps.leaf_algo == fil::leaf_algo_t::GROVE_PER_CLASS)
```

Comment:
Are there any python level classes that do multiclass predict_proba and compare directly against xgboost prediction output for those?

Given that the tests are nontrivial I think that would help.

Reply:
we certainly plan to add the python level tests. The ones on my desktop pass, but I haven't exhausted all possible test locations, so I was planning to have a separate PR for python, to simplify review. What do we do with this PR?